### PR TITLE
Update papers.md

### PR DESCRIPTION
**Pull request by eytan**
Some updates to published 2022 and 2023 papers.  Still a few more left to add.

## Motivation

(Write your motivation here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/pytorch/botorch/blob/main/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/pytorch/botorch, and link to your PR here.)

### CODE REVIEW

File: docs/papers.md [hunk 1]
```diff
@@ -18,9 +18,27 @@ The main reference for BoTorch is
 
 Here is an incomplete selection of peer-reviewed Bayesian optimization papers that build off of BoTorch:
 
+- [Discovering Many Diverse Solutions with Bayesian Optimization](https://arxiv.org/pdf/2210.10953.pdf). Natalie Maus, Kaiwen Wu, David Eriksson, Jacob Gardner. AISTATS 2023.
+
+- [Inducing Point Allocation for Sparse Gaussian Processes in High-Throughput Bayesian Optimisation](https://arxiv.org/pdf/2301.10123.pdf). Henry B. Moss, Sebastian W. Ober, Victor Picheny. AISTATS 2023.
+
+- [The Impact of Expertise in the Loop for Exploring Machine Rationality](https://arxiv.org/pdf/2302.05665.pdf).
```
Comment (esantorella) [inline docs/papers.md:25]:
```suggestion
- [The Impact of Expertise in the Loop for Exploring Machine Rationality](https://arxiv.org/pdf/2302.05665.pdf). Changkun Ou, Sven Mayer, Andreas Martin Butz. IUI 2023.
```

File: docs/papers.md [hunk 2]
```diff
@@ -18,9 +18,27 @@ The main reference for BoTorch is
...
+
+- [Local Latent Space Bayesian Optimization over Structured Inputs](https://proceedings.neurips.cc/paper_files/paper/2022/hash/ded98d28f82342a39f371c013dfb3058-Abstract-Conference.html). Natalie Maus, Haydn Jones, Juston Moore, Matt J. Kusner, John Bradshaw, Jacob Gardner. NeurIPS 2022.
+
+- [SnAKe: Bayesian Optimization via Pathwise Exploration](https://proceedings.neurips.cc/paper_files/paper/2022/hash/e4bab1843c8d5a69f5abfd0824593493-Abstract-Conference.html). Jose Pablo Folch, Shiqiang Zhang, Robert Lee, Behrang Shafei, David Walz, Calvin Tsay, Mark van der Wilk, Ruth Misener. NeurIPS 2022.
+
 - [Bayesian Optimization over Discrete and Mixed Spaces via Probabilistic Reparameterization](https://arxiv.org/pdf/2210.10199.pdf). Samuel Daulton, Xingchen Wan, David Eriksson, Maximilian Balandat, Michael A Osborne, Eytan Bakshy. NeurIPS 2022.
+
+- [Rethinking Optimization with Differentiable Simulation from a Global Perspective](https://proceedings.mlr.press/v205/antonova23a.html)Rika Antonova, Jingyun Yang, Krishna Murthy Jatavallabhula, Jeannette Bohg. CoRL 2022.
```
Comment (esantorella) [inline docs/papers.md:35]:
```suggestion
- [Rethinking Optimization with Differentiable Simulation from a Global Perspective](https://proceedings.mlr.press/v205/antonova23a.html). Rika Antonova, Jingyun Yang, Krishna Murthy Jatavallabhula, Jeannette Bohg. CoRL 2022.
```

File: docs/papers.md (hunk 1 above)
Comment (Balandat) [inline docs/papers.md:25]: Authors?

**Review by esantorella — APPROVED** [review]

**Comment by eytan** [comment]
OK, I've updated to include most of the recent papers I could find.  Can somebody please help w/ merging in?

**Pull request closed without merging**